### ISSUE TITLE
TypeScript Migration Plan

### ISSUE THREAD

**Issue by aaronlippold** [mitre/saf#1]
**Utility Functions**
* generate_map
* generate_ckl_metadata ( inspec2ckl )
* generate_inspec_metadata

**Exporters in Heimdall** 
***OCLIF CLI***
* inspec2ckl -> hdf2ckl
* inspec2csv -> hdf2csv
* inspec2xccdf -> hdf2xccdf

pdf2inspec ( not maintained) - Drop and redirect to python tool

**Merge these - add flag for CSV/XLS type - CIS or DISA**
* csv2inspec
* xlsx2inspec

**hdf-utils** PIPELINE ( Heimdall of the CICD pipeline ) computer see the results
***OCLIF CLI***
* compliance -> validate-threshold
* summary -> validation-summary

* New CLI SAF tool *

** Validation
- `saf create-profile csv2inspec -i file -o <path> --prefix (opt)` ( old inspec-tools)

** Normalization
- `saf convert nessus2hdf --i --o --mapping=file --prefix` ( old heimdall tools )
- `saf convert hdf2asff -i -o`
- `saf convert hdf2ckl -i -o --metadata file `

** Visualization
- `saf view` or `saf visualize -p xxxx ( default: 8080)`
- `saf view threshold`
- `saf view summary`

** Installers ( nodeJS + xxx ) MSI Wrapper or some fraamework to make multi-platform installers 
**** Driver: Approved Software List and Version ....
- WIndows 
- OSX
- Docker
- Linux (dpkg, dnf/rpm )

**Comment by ejaronne** [comment]
Initial reaction from a sponsor: don't replace one problem with another, i.e., heimdall_tools and inspec_tools confused the community because they thought heimdall_tools = heimdall, and inspec_tools = inspec. If we switch to "saf", we'd want a path to all of our saf tech via the saf tool. For example, an initial assumption was that we'd also have something like:

"saf scan" - which would run an inspec profile
or
"saf harden" - which would apply a hardening script.

In other words, saf as a one-stop shopping. I realize it's logistically hard to "wrap" ansible, terrform, inspec tools, etc. Instead, perhaps:

when someone types "saf scan", the command-line responds "visit https://saf.mitre.org/#/validate to explore and run inspec profiles"

when someone types "saf harden", the command-line responds "visit https://saf.mitre.org/#/harden to explore and run hardening scripts"

**Comment by rbclark** [comment]
Adding `saf scan` and `saf harden` responses from the SAF utility would not a problem.
